pkg/routeagent_driver/handlers/kubeproxy_iptables/endpoint_handler.go

```diff
@@ -0,0 +1,187 @@
+
+			kp.vxlanDevice = nil
+		}
+
+		kp.isGatewayNode = false
+		localClusterGwNodeIP := net.ParseIP(endpoint.Spec.PrivateIP)
+		remoteVtepIP, err := kp.getVxlanVtepIPAddress(localClusterGwNodeIP.String())
+		if err != nil {
+			return fmt.Errorf("failed to derive the remoteVtepIP %v", err)
+		}
+
+		klog.Infof("Creating the vxlan interface %s with gateway node IP %s", VxLANIface, localClusterGwNodeIP)
+		err = kp.createVxLANInterface(endpoint.Spec.Hostname, VxInterfaceWorker, localClusterGwNodeIP)
+		if err != nil {
+			klog.Fatalf("Unable to create VxLAN interface on non-GatewayNode (%s): %v", endpoint.Spec.Hostname, err)
+		}
+
+		kp.vxlanGwIP = &remoteVtepIP
+		err = kp.reconcileRoutes(remoteVtepIP)
+		if err != nil {
+			return fmt.Errorf("error while reconciling routes %v", err)
+		}
+	}
+
+	return nil
+}
+
+func (kp *SyncHandler) LocalEndpointUpdated(endpoint *submV1.Endpoint) error {
+	return nil
+}
+
+func (kp *SyncHandler) LocalEndpointRemoved(endpoint *submV1.Endpoint) error {
+	kp.syncHandlerMutex.Lock()
+	defer kp.syncHandlerMutex.Unlock()
+	kp.isGatewayNode = false
+
+	// If the vxLAN device exists and it points to the same endpoint, delete it.
+	if kp.vxlanDevice != nil && kp.vxlanDevice.activeEndpointHostname == endpoint.Spec.Hostname {
+		err := kp.vxlanDevice.deleteVxLanIface()
+		kp.vxlanDevice = nil
+		kp.vxlanGwIP = nil
+		if err != nil {
+			return fmt.Errorf("failed to delete the the vxlan interface on Endpoint removal: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func (kp *SyncHandler) RemoteEndpointCreated(endpoint *submV1.Endpoint) error {
+	if err := kp.overlappingSubnets(endpoint.Spec.Subnets); err != nil {
+		// Skip processing the endpoint when CIDRs overlap and return nil to avoid re-queuing.
+		klog.Errorf("overlappingSubnets for new remote %#v returned error: %v", endpoint, err)
+		return nil
+	}
+
+	kp.syncHandlerMutex.Lock()
+	defer kp.syncHandlerMutex.Unlock()
+	for _, inputCidrBlock := range endpoint.Spec.Subnets {
+		if !kp.remoteSubnets.Contains(inputCidrBlock) {
+			kp.remoteSubnets.Add(inputCidrBlock)
+		}
+	}
+
+	if err := kp.updateRoutingRulesForInterClusterSupport(endpoint.Spec.Subnets, Add); err != nil {
+		klog.Errorf("updateRoutingRulesForInterClusterSupport for new remote %#v returned error: %+v",
+			endpoint, err)
+		return err
+	}
+	// Add routes to the new endpoint on the GatewayNode.
+	kp.updateRoutingRulesForHostNetworkSupport(endpoint.Spec.Subnets, Add)
+	kp.updateIptableRulesForInterClusterTraffic(endpoint.Spec.Subnets, Add)
+
+	return nil
+}
+
+func (kp *SyncHandler) RemoteEndpointUpdated(endpoint *submV1.Endpoint) error {
+	return nil
+}
+
+func (kp *SyncHandler) RemoteEndpointRemoved(endpoint *submV1.Endpoint) error {
+	kp.syncHandlerMutex.Lock()
+	defer kp.syncHandlerMutex.Unlock()
+	for _, inputCidrBlock := range endpoint.Spec.Subnets {
+		if !kp.remoteSubnets.Contains(inputCidrBlock) {
+			kp.remoteSubnets.Delete(inputCidrBlock)
+		}
+	}
+	// TODO: Handle a remote endpoint removal use-case
```

Comment:
I'm my case (ovn) I'm reading the system state, and calculating the diff or rules to set.